bitcoin/src/blockdata/witness.rs

```diff
@@ -55,16 +57,13 @@ impl Decodable for Witness {
             Ok(Witness::default())
         } else {
             let mut cursor = 0usize;
-            let mut last = 0usize;
-            let mut second_to_last = 0usize;
 
             // this number should be determined as high enough to cover most witness, and low enough
             // to avoid wasting space without reallocating
             let mut content = vec![0u8; 128];
+            let mut indices = Vec::with_capacity(witness_elements * U32_SIZE);
```

Comment:
If we do this, we should just don't bother to encode the indeces in content and use it as a field.

I think we should drop push

Reply:
If we decide to keep `push` I added more to the function `test_push` to convince myself the logic was correct, feel free to use
```
        witness.push(&vec![4u8, 5u8]);
        let expected = Witness {
            witness_elements: 3,
            content: vec![1u8, 0, 2, 2, 3, 2, 4, 5, 0, 0, 0, 0, 2, 0, 0, 0, 5, 0, 0, 0],
            indices_start: 8,
        };
        assert_eq!(witness, expected);
        assert_eq!(witness.last(), Some(&[4u8, 5u8][..]));
        assert_eq!(witness.second_to_last(), Some(&[2u8, 3u8][..]));
```

Reply:
@RCasatta  I have removed the extra allocation and replaced it with a rotate_left at the end.

@tcharding Thanks I added that to the tests.